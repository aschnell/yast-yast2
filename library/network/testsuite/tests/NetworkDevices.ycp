{

include "testsuite.ycp";

map READ = $[
];

TESTSUITE_INIT([READ], nil);
import "NetworkDevices";

DUMP("device_type");
TEST(``(NetworkDevices::device_type("eth0")), [], nil);
TEST(``(NetworkDevices::device_type("eth1")), [], nil);
TEST(``(NetworkDevices::device_type("eth-pcmcia-0")), [], nil);
TEST(``(NetworkDevices::device_type("eth-usb-1")), [], nil);
TEST(``(NetworkDevices::device_type("tr-pcmcia-1")), [], nil);
TEST(``(NetworkDevices::device_type("ippp2")), [], nil);
TEST(``(NetworkDevices::device_type("ppp2")), [], nil);
TEST(``(NetworkDevices::device_type("lo")), [], nil);
TEST(``(NetworkDevices::device_type("eth0#1")), [], nil);
TEST(``(NetworkDevices::device_type("eth1#20")), [], nil);
TEST(``(NetworkDevices::device_type("eth-pcmcia-0#3")), [], nil);
TEST(``(NetworkDevices::device_type("eth-usb-1#0")), [], nil);
TEST(``(NetworkDevices::device_type("tr-pcmcia-2#432")), [], nil);
TEST(``(NetworkDevices::device_type("ippp2#2")), [], nil);
TEST(``(NetworkDevices::device_type("ppp2#2")), [], nil);
TEST(``(NetworkDevices::device_type("lo#1")), [], nil);
TEST(``(NetworkDevices::device_type("eth1#blah")), [], nil);

DUMP("device_num");
TEST(``(NetworkDevices::device_num("eth0")), [], nil);
TEST(``(NetworkDevices::device_num("eth1")), [], nil);
TEST(``(NetworkDevices::device_num("eth-pcmcia-0")), [], nil);
TEST(``(NetworkDevices::device_num("eth-usb-1")), [], nil);
TEST(``(NetworkDevices::device_num("tr-pcmcia-2")), [], nil);
TEST(``(NetworkDevices::device_num("ippp2")), [], nil);
TEST(``(NetworkDevices::device_num("ppp2")), [], nil);
TEST(``(NetworkDevices::device_num("eth0#1")), [], nil);
TEST(``(NetworkDevices::device_num("eth1#20")), [], nil);
TEST(``(NetworkDevices::device_num("eth-pcmcia-0#3")), [], nil);
TEST(``(NetworkDevices::device_num("eth-usb-1#0")), [], nil);
TEST(``(NetworkDevices::device_num("tr-pcmcia-2#432")), [], nil);
TEST(``(NetworkDevices::device_num("ippp2#2")), [], nil);
TEST(``(NetworkDevices::device_num("ppp2#2")), [], nil);
TEST(``(NetworkDevices::device_num("eth1#blah")), [], nil);
TEST(``(NetworkDevices::device_num("lo#1")), [], nil);
TEST(``(NetworkDevices::device_num("lo")), [], nil);

/*
DUMP("alias_num");
TEST(``(NetworkDevices::alias_num("eth0#1")), [], nil);
TEST(``(NetworkDevices::alias_num("eth1#20")), [], nil);
TEST(``(NetworkDevices::alias_num("eth-pcmcia-0#3")), [], nil);
TEST(``(NetworkDevices::alias_num("eth-usb-1#0")), [], nil);
TEST(``(NetworkDevices::alias_num("tr-pcmcia-2#432")), [], nil);
TEST(``(NetworkDevices::alias_num("ippp2#2")), [], nil);
TEST(``(NetworkDevices::alias_num("ppp2#2")), [], nil);
TEST(``(NetworkDevices::alias_num("lo#1")), [], nil);
TEST(``(NetworkDevices::alias_num("eth0#blah")), [], nil);
TEST(``(NetworkDevices::alias_num("eth0")), [], nil);
TEST(``(NetworkDevices::alias_num("lo")), [], nil);
*/

DUMP("device_name");
TEST(``(NetworkDevices::device_name("eth","0")), [], nil);
TEST(``(NetworkDevices::device_name("eth","1")), [], nil);
TEST(``(NetworkDevices::device_name("eth-pcmcia-","0")), [], nil);
TEST(``(NetworkDevices::device_name("tr-pcmcia-","1")), [], nil);
TEST(``(NetworkDevices::device_name("ippp","2")), [], nil);
TEST(``(NetworkDevices::device_name("ppp","2")), [], nil);
TEST(``(NetworkDevices::device_name("lo","")), [], nil);

/*
DUMP("alias_name");
TEST(``(NetworkDevices::alias_name("eth","0","5")), [], nil);
TEST(``(NetworkDevices::alias_name("eth","1","4")), [], nil);
TEST(``(NetworkDevices::alias_name("eth-pcmcia-","0","1")), [], nil);
TEST(``(NetworkDevices::alias_name("tr-pcmcia-","1","1")), [], nil);
TEST(``(NetworkDevices::alias_name("ippp","2","0")), [], nil);
TEST(``(NetworkDevices::alias_name("ppp","2","0")), [], nil);
TEST(``(NetworkDevices::alias_name("lo", "", "7")), [], nil);
*/

}
