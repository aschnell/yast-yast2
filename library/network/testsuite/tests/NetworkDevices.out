Dump	Combined:
Dump	ifcfg-eth0
Dump	["eth", "0", ""]
Dump	ifcfg-eth1
Dump	["eth", "1", ""]
Dump	ifcfg-eth-pcmcia-0
Dump	["eth", "pcmcia-0", ""]
Dump	ifcfg-eth-usb-1
Dump	["eth", "usb-1", ""]
Dump	ifcfg-tr-pcmcia-1
Dump	["tr", "pcmcia-1", ""]
Dump	ifcfg-ippp2
Dump	["ippp", "2", ""]
Dump	ifcfg-ppp2
Dump	["ppp", "2", ""]
Dump	ifcfg-lo
Dump	["lo", "", ""]
Dump	ifcfg-eth0#1
Dump	["eth", "0", "1"]
Dump	ifcfg-eth1#20
Dump	["eth", "1", "20"]
Dump	ifcfg-eth-pcmcia-0#3
Dump	["eth", "pcmcia-0", "3"]
Dump	ifcfg-eth-usb-1#0
Dump	["eth", "usb-1", "0"]
Dump	ifcfg-tr-pcmcia-2#432
Dump	["tr", "pcmcia-2", "432"]
Dump	ifcfg-ippp2#2
Dump	["ippp", "2", "2"]
Dump	ifcfg-ppp2#2
Dump	["ppp", "2", "2"]
Dump	ifcfg-lo#1
Dump	["lo", "", "1"]
Dump	ifcfg-eth1#blah
Dump	["eth", "1", "blah"]
Dump	ifcfg-eth-id-00:07:e9:d5:8e:e8
Dump	["eth", "id-00:07:e9:d5:8e:e8", ""]
Dump	ifcfg-eth-id-00:07:e9:d5:8e:e8#foo
Dump	["eth", "id-00:07:e9:d5:8e:e8", "foo"]
Dump	ifcfg-ip6tnl0
Dump	["ip6tnl", "0", ""]
Dump	ifcfg-ip6tnl31
Dump	["ip6tnl", "31", ""]
Dump	ifcfg-mip6mnha3
Dump	["mip6mnha", "3", ""]
Dump	Obsolete:
Return	eth-pcmcia-0
Return	tr-pcmcia-1
Return	eth-pcmcia-0#1
Return	tr-pcmcia-1#1
